lib/rspec/mocks.rb

```diff
@@ -101,6 +101,23 @@ def self.with_temporary_scope
       end
     end
 
+    @@excluded_subclasses = []
+
+    def self.excluded_subclasses
+      @@excluded_subclasses.select(&:weakref_alive?).map(&:__getobj__)
```

Comment:
This could be compacted to prevent endless growth.

Reply:
I added `.compact`.